config/accounts.go

```diff
@@ -39,20 +44,147 @@ func (a *AccountCCPA) EnabledForIntegrationType(integrationType IntegrationType)
+	Purpose2                   AccountGDPRPurpose `mapstructure:"purpose2" json:"purpose2"`
+	Purpose3                   AccountGDPRPurpose `mapstructure:"purpose3" json:"purpose3"`
+	Purpose4                   AccountGDPRPurpose `mapstructure:"purpose4" json:"purpose4"`
+	Purpose5                   AccountGDPRPurpose `mapstructure:"purpose5" json:"purpose5"`
+	Purpose6                   AccountGDPRPurpose `mapstructure:"purpose6" json:"purpose6"`
+	Purpose7                   AccountGDPRPurpose `mapstructure:"purpose7" json:"purpose7"`
+	Purpose8                   AccountGDPRPurpose `mapstructure:"purpose8" json:"purpose8"`
+	Purpose9                   AccountGDPRPurpose `mapstructure:"purpose9" json:"purpose9"`
+	Purpose10                  AccountGDPRPurpose `mapstructure:"purpose10" json:"purpose10"`
+	// Hash table of purpose configs for convenient purpose config lookup
+	PurposeConfigs      map[consentconstants.Purpose]*AccountGDPRPurpose
+	PurposeOneTreatment AccountGDPRPurposeOneTreatment `mapstructure:"purpose_one_treatment" json:"purpose_one_treatment"`
+	SpecialFeature1     AccountGDPRSpecialFeature      `mapstructure:"special_feature1" json:"special_feature1"`
 }
 
 // EnabledForIntegrationType indicates whether GDPR is turned on at the account level for the specified integration type
-// by using the integration type setting if defined or the general GDPR setting if defined; otherwise it returns nil
+// by using the integration type setting if defined or the general GDPR setting if defined; otherwise it returns nil.
 func (a *AccountGDPR) EnabledForIntegrationType(integrationType IntegrationType) *bool {
-
 	if integrationEnabled := a.IntegrationEnabled.GetByIntegrationType(integrationType); integrationEnabled != nil {
 		return integrationEnabled
 	}
 	return a.Enabled
 }
+// PurposeEnforced checks if full enforcement is turned on for a given purpose at the account level. It returns the 
+// enforcement strategy type and whether or not it is set on the account. If not set, a default value of true is 
+// returned matching host default behavior.
+func (a *AccountGDPR) PurposeEnforced(purpose consentconstants.Purpose) (value, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return true, false
+	}
+	if a.PurposeConfigs[purpose].EnforcePurpose == TCF2FullEnforcement {
+		return true, true
+	}
+	if a.PurposeConfigs[purpose].EnforcePurpose == TCF2NoEnforcement {
+		return false, true
+	}
+	return true, false
+}
+
+// PurposeEnforcingVendors gets the account level enforce vendors setting for a given purpose returning the value and
+// whether or not it is set. If not set, a default value of true is returned matching host default behavior.
+func (a *AccountGDPR) PurposeEnforcingVendors(purpose consentconstants.Purpose) (value, exists bool) {
+	if a.PurposeConfigs[purpose] == nil {
+		return true, false
+	}
+	if a.PurposeConfigs[purpose].EnforceVendors == nil {
+		return true, false
+	}
```

Comment:
Another nitpick:
```
87   // PurposeEnforcingVendors gets the account level enforce vendors setting for a given purpose returning the value and
88   // whether or not it is set. If not set, a default value of true is returned matching host default behavior.
89   func (a *AccountGDPR) PurposeEnforcingVendors(purpose consentconstants.Purpose) (value, exists bool) {
90 -     if a.PurposeConfigs[purpose] == nil {
91 -         return true, false
92 -     }
93 -     if a.PurposeConfigs[purpose].EnforceVendors == nil {
   +     if a.PurposeConfigs[purpose] == nil || a.PurposeConfigs[purpose].EnforceVendors == nil {
94           return true, false
95       }
96       return *a.PurposeConfigs[purpose].EnforceVendors, true
97   }
98   
config/accounts.go
```

Reply:
See my thoughts here: https://github.com/prebid/prebid-server/pull/2146#discussion_r819808602. We can discuss offline.